tika-parsers/src/main/java/org/apache/tika/parser/pkg/RarParser.java

```diff
@@ -86,7 +86,7 @@ public void parse(InputStream stream, ContentHandler handler,
                         subFile = rar.getInputStream(header);
 
                         Metadata entrydata = PackageParser.handleEntryMetadata(
-                                header.getFileNameString(),
+                                header.getFileNameW().equals("")?header.getFileNameString():header.getFileNameW(),

```

Comment:
you said is better, but member variables should not be null , Come to this issue , What i mean is that var header has two methods 1. getFileNameString()    2. getFileNameW()  . Your unit test case just cover iso-8859-1 encoding 
tip:https://github.com/apache/tika/blob/trunk/tika-parsers/src/test/resources/test-documents/test-documents.rar
i have tested for chinese and other language , it didn't work , please 
try to debug for this and you will find you have to check for the charset of the fieName.